src/state_db.rs

```diff
@@ -360,22 +360,23 @@ impl StateDB {
     }
     fn proof(ctx: &mut FunctionContext) -> NeonResult<smt::Proof> {
         let raw_proof = ctx.argument::<JsObject>(2)?;
-        let mut sibling_hashes = NestedVec::new();
         let raw_sibling_hashes = raw_proof
             .get::<JsArray, _, _>(ctx, "siblingHashes")?
             .to_vec(ctx)?;
-        for key in raw_sibling_hashes.iter() {
-            let key = key
-                .downcast_or_throw::<JsTypedArray<u8>, _>(ctx)?
-                .as_slice(ctx)
-                .to_vec();
-            sibling_hashes.push(key);
-        }
+        let sibling_hashes = raw_sibling_hashes
+            .iter()
+            .map(|key| {
+                Ok(key
+                    .downcast_or_throw::<JsTypedArray<u8>, _>(ctx)?
+                    .as_slice(ctx)
+                    .to_vec())
+            })
+            .collect::<NeonResult<Vec<Vec<u8>>>>()?;
```

Comment:
We can use `NestedVec` instead of `Vec<Vec<u8>>`

Reply:
Done.